core/platform/email/mailgun_email_services.py

```diff
@@ -77,8 +85,13 @@ def send_email_to_recipients(
     Returns:
         bool. Whether the emails are sent successfully.
     """
-    if not feconf.MAILGUN_API_KEY:
-        raise Exception('Mailgun API key is not available.')
+    mailgun_api_key: Optional[str] = secrets_services.get_secret(
+        'MAILGUN_API_KEY')
+    if mailgun_api_key is None:
+        logging.exception('Cloud Secret Manager is not working.')
```

Comment:
Here and elsewhere, maybe you could have a clearer log -- something like "Cloud Secret Manager returned None for mailgun_api_key"

Reply:
Done.